pubsub/google/cloud/pubsub_v1/publisher/batch/thread.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2017, Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+
+import threading
+import time
+
+from google.cloud.pubsub_v1 import types
+from google.cloud.pubsub_v1.publisher import exceptions
+from google.cloud.pubsub_v1.publisher import futures
+from google.cloud.pubsub_v1.publisher.batch import base
+
+
+class Batch(base.Batch):
```

Comment:
I am somewhat concerned that there's no locks in this class. What happens if a monitor thread and another thread call `commit`? Will this try to send the batch twice? It seems like you might want to use some sort of locking primitive to make sure that commit happens once and only once.